testing/modules/gpregression_test.py

```diff
@@ -202,5 +185,47 @@ def test_sampling_prediction(self):
         gp.gp_predict.jitter = 1e-6
 
         y_samples = infr_pred.run(X=mx.nd.array(Xt, dtype=dtype))[0].asnumpy()
-
         # TODO: Check the correctness of the sampling
+
+    def test_with_samples(self):
+        from mxfusion.common import config
+        config.DEFAULT_DTYPE = 'float64'
+        dtype = 'float64'
+
+        D, X, Y, noise_var, lengthscale, variance = self.gen_data()
+
+        m = Model()
```

Comment:
Is there a reason you don't use gen_mxfusion_model here? Same for the other new sample tests.